rest/model/distro.go

```diff
@@ -18,6 +22,20 @@ func (apiDistro *APIDistro) BuildFromService(h interface{}) error {
 	case distro.Distro:
 		apiDistro.Name = ToAPIString(v.Id)
 		apiDistro.UserSpawnAllowed = v.SpawnAllowed
+		apiDistro.Provider = ToAPIString(v.Provider)
+
+		if v.ProviderSettings != nil {
```

Comment:
we must also check if provider == ec2 before attempting this branch

Reply:
Done